### ISSUE TITLE
Objects in a scene are 'Awaking' after InstanceFinder.SceneManager.OnLoadEnd is Invoked

### ISSUE THREAD

**Issue by tkuebler** [FirstGearGames/FishNet#70]
Unity:     2021.3.5f1
Fishnet:  Pro 2.2.2
Running in Editor in Host mode.

Steps To Reproduce:  

1) upgrade to 2.2.2 pro
2) Place a monobehavior script in a scene with a print statement in it's awake method
3) put a print statement at the top of a registered callback to InstanceFinder.SceneManager.OnLoadEnd
4) observe the order of print statements in the editor when it runs - onloadend -> awake
5) Downgrade to 2.2.1 pro
6) Observe the order of print statements in the editor when it runs  - awake -> onloadend

I'm discovering spawn points via FindObjectsOfType in each new scene and moving players to them - this has been working as expected since I've been using fishnet.  

FindObjectsOfType is returning null in 2.2.2 pro OnLoadEnd callback even if I set includeInactive = true in the call.  So that infers that when OnLoadEnd is called the game objects are not in the scene yet or the scene isn't actually loaded.

**Comment by FirstGearGames** [comment]
The objects you're searching for are monobehaviours?

**Comment by FirstGearGames** [comment]
Resolved in 2.2.3.

Disregard my last comment. I added something in for this but looks like it didn't make it to release. I'll push a new release real quick... here's the comments.

```
            /* Must yield after sceneProcessor handles things.
            * This is a Unity bug of sorts. I'm not entirely sure what
            * is happening, but without the yield it seems as though
            * the processor logic doesn't complete. This doesn't make much
            * sense given unity is supposed to be single threaded. Must be
            * something to do with the coroutine. */
            yield return null;
```
This is added before each SetActiveScene(); call.

**Comment by tkuebler** [comment]
> The objects you're searching for are monobehaviours?
yes
> Also is it returning null or empty, because it should never return null.
empty, yes, sorry.  the null is when I access the first element without checking the size like a moron... ;)
>
> Are you getting objects from the new scene, from a different scene?
the new scene that is supposedly loaded.

Thanks for the fix!!
